include/zephyr/ap_pwrseq/ap_pwrseq.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef _AP_PWRSEQ_H_
+#define _AP_PWRSEQ_H_
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AP_POWER_STATE_ACTION_ENTRY	BIT(0)
+#define AP_POWER_STATE_ACTION_RUN	BIT(1)
+#define AP_POWER_STATE_ACTION_EXIT	BIT(2)
+
+#define AP_POWER_SUB_STATE_ENUM_DEF_WITH_COMMA(node_id, prop, idx)            \
+	DT_CAT6(node_id, _P_, prop, _IDX_, idx, _UPPER_TOKEN),
+
+#define AP_PWRSEQ_EACH_CHIPSET_SUB_STATE_ENUM_DEF(node_id)                    \
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, chipset),                       \
+		(DT_FOREACH_PROP_ELEM(node_id, chipset,                       \
+			AP_POWER_SUB_STATE_ENUM_DEF_WITH_COMMA)),             \
+		())
+
+#define AP_PWRSEQ_EACH_APP_SUB_STATE_ENUM_DEF(node_id)                        \
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, application),                   \
+		(DT_FOREACH_PROP_ELEM(node_id, application,                   \
+			AP_POWER_SUB_STATE_ENUM_DEF_WITH_COMMA)),             \
+		())
+
+/** @brief AP power sequence valid power states. */
+enum ap_pwrseq_state {
```

Comment:
add a psudo G3 also,

Reply:
I am not aware of what this pseudo state is, is it UNINIT?

Reply:
/* Both EC and AP are off */
AP_POWER_STATE_PG3